superset-frontend/src/filters/components/Select/AntdSelectFilter.tsx

```diff
@@ -76,7 +76,10 @@ export default function AntdPluginFilterSelect(
       ),
       // @ts-ignore (add to superset-ui/core)
       currentState: {
-        value: resultValue,
+        value:
+          Array.isArray(resultValue) && resultValue.length === 0
```

Comment:
I think `resultValue` always is array according line 60

Reply:
Oh nice catch! I was getting `value` and `resultValue` mixed up here.